Baekjoon/2231_1.py

```diff
@@ -2,25 +2,38 @@
 문제 링크:https://www.acmicpc.net/problem/2231
 카테고리:브루트포스
 문제해설:분해합의 생성자 구하기
+
+M의 각자리수 + M
+
+*523%10=3 append->(523-3)//10=52
+->*52%10=2 append-> (52-2)//10=5=b
+->if b<10->break
+
+216입력->198출력
 '''
-n=int(input())
 
-def func(m):
+N=int(input())
+
+def M_sum(M):
   arr=[]
-  a=m%10
-  arr.append(m)
-  m=(m-a)%10
-  print(arr)
-  if m!=0:
-    func(m)
-  elif m==0:
-    m=(m-a)//10
-    arr.append(m)
-
-for m in range(1,n+1):
-  m_sum=m+sum(func(m))
-  if m_sum==n:
-    print(m)
-    break
-  else:
-    print(0)
+
+  while True:
+    a=M%10
+    arr.append(a)
+    M=(M-a)//10
+  
+    if M<10:
+      arr.append(M)
+      break
+
+  return sum(arr)
```

Comment:
```suggestion
  ret = M

  while M > 0:
    ret += M%10
    M /= 10
  
  return ret
```

arr 을 안쓰고 while문에서 합계를 누적할 수  있고,
while의 조건에 탈출 조건을 같이 쓸 수 있어요

처음에 M을 결과값을 더해주면 L33 에서 `M + M_sum` 대신 M_sum 함수 자체로 판단가능하겠습니다

Reply:
L23 `M /= 10`->`M //= 10`
로 하니까 잘 나옵니당~!